src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -2065,6 +2068,7 @@ public DebuggerCommandResults Invoke(ManualResetEventSlim startInvokeEvent)
                 }
                 catch (System.Security.SecurityException) { }
                 _identityToImpersonate = ((currentIdentity != null) && (currentIdentity.ImpersonationLevel == TokenImpersonationLevel.Impersonation)) ? currentIdentity : null;
+#endif
```

Comment:
Another option is to only keep this if/def pair and remove the rest.
Referencing to `WindowsIdentity` is OK on Unix platform as long as we don't call methods on it (only then will the `PlatformNotSupportedException` be thrown).
This pair of if/def will make `_identityToImpersonate` always `null` on Unix, and thus we won't get in the code path that access any members of the `WindowsIdentity` type.

Reply:
I am not sure what you mean, but I prefer to remove everything from UNIX builds that isn't used there.  I don't like having dead code paths...